microsoft/testsuites/core/storage.py

```diff
@@ -0,0 +1,203 @@
+    )
+
+    # /dev/sda1 on / type ext4 (rw,relatime,discard)
+    _root_partition_regex = re.compile(
+        r"\s*\/dev\/(?P<partition>\D+).*\s+on\s+\/\s+type.*"
+    )
+
+    @TestCaseMetadata(
+        description="""
+        This test will check that VM root disk(os disk) is provisioned
+        with the correct timeout.
+        Steps:
+        1. Find the root disk (os disk) partition for the VM. The root partition
+        corresponds to entry with mount point `/' in the `mount` command.
+        2. Verify the timeout value for root disk in
+        `/sys/block/<partition>/device/timeout` file is set to 300.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+        ),
+    )
+    def verify_root_device_timeout(
+        self,
+        log: Logger,
+        node: RemoteNode,
+    ) -> None:
+        os_disk_partition = self._get_os_disk_partition(log, node)
+        output = int(
+            node.tools[Cat].run(f"/sys/block/{os_disk_partition}/device/timeout").stdout
+        )
+        assert_that(output).is_equal_to(self.DEFAULT_DISK_TIMEOUT)
+
+    @TestCaseMetadata(
+        description="""
+        This test will check that the resource disk is present in the list of mounted
+        devices. VMs contain a resource disk, which is not a managed disk. The
+        resource disk provides short-term storage for applications and processes, and
+        is intended to only store data such as page or swap files.
+        Steps:
+        1. Get the mount point for the resource disk. If `/var/log/cloud-init.log`
+        file is present, mount location is `\\mnt`, otherwise it is obtained from
+        `ResourceDisk.MountPoint` entry in `waagent.conf` configuration file.
+        2. Get the resource disk partition. In a VM with no data disk, one of `sda`
+        or `sdb` is a resource disk partition. The root partition corresponds to
+        entry with mount point `/' in the `mount` command, and the other is
+        resource disk.
+        3. Verify that "/dev/<resource_disk_partition> <mount_point>` entry is
+        present in `/etc/mtab` file.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            environment_status=EnvironmentStatus.Deployed,
+        ),
+    )
+    def verify_resource_disk_mtab_entry(self, log: Logger, node: RemoteNode) -> None:
+        resource_disk_mount_point = self._get_data_disk_mount_point(log, node)
+        os_disk_partition = self._get_os_disk_partition(log, node)
+        if os_disk_partition == "sda":
+            mntresource = "/dev/sdb1 " + resource_disk_mount_point
+        else:
+            mntresource = "/dev/sda1 " + resource_disk_mount_point
+
+        output = node.tools[Cat].run("/etc/mtab").stdout
+        if mntresource in output:
+            log.debug("ResourceDisk entry is present.")
+        else:
+            raise LisaException("ResourceDisk entry is not present.")
```

Comment:
It needs to output the expected value. The command output is in debug log already, but the expected mntresource is not printed. If it's failed, we need to know what's expected in the code logic.